addons/sourcemod/scripting/include/Spectate.inc

```diff
@@ -8,6 +8,21 @@
 #endif
 #define _Spectate_included
 
+/**
+ * Called right before a client was force suicide before moving to spec
+ */
+forward void Spectate_OnCommitSuicide(int client);
+
+/**
+ * Called right after a client was moving to spec
+ */
+forward void Spectate_OnClientMovedToSpec(int client);
```

Comment:
you can hook player_team with HookEvent to know this:
- https://sourcemod.dev/#/events/function.HookEvent
- https://wiki.alliedmods.net/Generic_Source_Events#player_team